nsv13/code/modules/power/stormdrive.dm

```diff
@@ -870,100 +906,318 @@ Control Rods
+	if(state == REACTOR_STATE_MELTDOWN)
+		icon_state = "broken"
+		return
+	if(state == REACTOR_STATE_REPAIR) //TEMP
+		icon_state = "broken"
+		return
+	if(state == REACTOR_STATE_REINFORCE) //TEMP
+		icon_state = "broken"
+		return
+	if(state == REACTOR_STATE_REINFORCE) //TEMP
+		icon_state = "broken"
+		return
+	cut_overlays()
+	if(can_cool()) //If control rods aren't destroyed.
+		switch(round(control_rod_percent)) //move the control rods up and down
+			if(0 to 24)
+				add_overlay("rods_[control_rods.len]_1")
+			if(25 to 49)
+				add_overlay("rods_[control_rods.len]_2")
+			if(50 to 74)
+				add_overlay("rods_[control_rods.len]_3")
+			if(75 to 100)
+				add_overlay("rods_[control_rods.len]_4")
+	if(state == REACTOR_STATE_MAINTENANCE)
+		icon_state = "reactor_maintenance" //If we're in maint, don't make it appear hot.
+		return
+	if(state == REACTOR_STATE_RUNNING)
+		icon_state = "reactor_on" //Default to being on.
+		check_meltdown_warning()
+		var/illumination = max(2, heat/100)
+		if(heat > 0 && heat <= reactor_temperature_nominal) //checking temperature ranges
+			icon_state = "reactor_on"
+			light_color = LIGHT_COLOR_CYAN
+			set_light(illumination)
+		else if(heat > reactor_temperature_nominal && heat <= reactor_temperature_hot)
+			icon_state = "reactor_hot"
+			light_color = LIGHT_COLOR_CYAN
+			set_light(illumination)
+		else if(heat > reactor_temperature_hot && heat <= reactor_temperature_critical)
+			icon_state = "reactor_hot" //NEED NEW ICON
+			light_color = LIGHT_COLOR_CYAN
+			set_light(illumination)
+		else if(heat > reactor_temperature_critical && heat <= reactor_temperature_meltdown) //Final Warning
+			icon_state = "reactor_overheat"
+			light_color = LIGHT_COLOR_RED
+			set_light(illumination)
+		else if(heat > reactor_temperature_meltdown)
+			icon_state = "reactor_overheat"
+			light_color = LIGHT_COLOR_RED
+			set_light(illumination)
+			start_meltdown() //Epsilon or Death
 
 /obj/machinery/atmospherics/components/binary/stormdrive_reactor/Destroy()
 	for(var/atom/X in contents)
 		qdel(X)
 	.=..()
 
-/obj/machinery/atmospherics/components/binary/stormdrive_reactor/proc/handle_gas_records()
-	if(world.time >= gas_records_next_interval)
-		gas_records_next_interval = world.time + gas_records_interval
+/obj/machinery/atmospherics/components/binary/stormdrive_reactor/proc/storm_zap(atom/source, zap_range = 3, power, tesla_flags = TESLA_DEFAULT_FLAGS, list/shocked_targets) //This is so EHHHHHH but CBS fixing
+	. = source.dir
+
+	var/closest_dist = 0
+	var/closest_atom
+	var/obj/machinery/power/tesla_coil/closest_tesla_coil
+	var/obj/machinery/power/grounding_rod/closest_grounding_rod
+	var/mob/living/closest_mob
+	var/obj/machinery/closest_machine
+	var/obj/structure/closest_structure
+	var/obj/structure/blob/closest_blob
+	var/static/things_to_shock = typecacheof(list(/obj/machinery, /mob/living, /obj/structure))
+	var/static/blacklisted_tesla_types = typecacheof(list(/obj/machinery/atmospherics,
```

Comment:
Why are you making things static in a proc? Static is for global member access by subclasses and is unnecessary as you're just redefining the static list over and over again with each usage of this proc

Reply:
(And again, put this as a field on a tesla_zap component so it can be modified meaningfully)